clients/src/test/java/org/apache/kafka/clients/consumer/internals/HeartbeatRequestManagerTest.java

```diff
@@ -746,6 +746,37 @@ public void testSendingLeaveGroupHeartbeatWhenPreviousOneInFlight(final short ve
         NetworkClientDelegate.PollResult pollAgain = heartbeatRequestManager.poll(time.milliseconds());
         assertEquals(0, pollAgain.unsentRequests.size());
     }
+    
+    @ParameterizedTest
+    @ApiKeyVersionsSource(apiKey = ApiKeys.CONSUMER_GROUP_HEARTBEAT)
+    public void testConsumerAcksReconciledAssignmentWithFirstHeartBeatAckLost(final short version) {
+        // 1. complete reconciliation
+        createHeartbeatStatAndRequestManager();
+        String topic = "topic1";
+        int exceededTimeMs = 5;
+        Set<String> set = Collections.singleton(topic);
+        when(subscriptions.subscription()).thenReturn(set);
+        subscriptions.subscribe(set, Optional.empty());
+        mockReconcilingMemberData();
+        // 2. send heartbeat1 to ack assignment tp0
+        time.sleep(DEFAULT_HEARTBEAT_INTERVAL_MS);
+        NetworkClientDelegate.PollResult result = heartbeatRequestManager.poll(time.milliseconds());
+        // 3. HB1 times out
+        result.unsentRequests.get(0)
+                .handler()
+                .onFailure(time.milliseconds(), new TimeoutException("timeout"));
+        // 4. heartbeat request manager resets the sentFields to null HeartbeatState.reset()
+        time.sleep(DEFAULT_MAX_POLL_INTERVAL_MS + exceededTimeMs);
```

Comment:
I would expect that just sleeping the interval would be enough, so we could maybe simplify the test and remove the `exceededTimeMs` from here and the var itself?